src/src/radio.c

```diff
@@ -3053,58 +3149,66 @@ static bool validatePacketTimestamp(RAIL_RxPacketDetails_t *pPacketDetails, uint
 static void updateRxFrameDetails(RAIL_RxPacketDetails_t *pPacketDetails,
                                  bool                    securedOutgoingEnhancedAck,
                                  bool                    framePendingSetInOutgoingAck,
-                                 uint8_t                 iid)
+                                 uint8_t                 iid,
+                                 uint8_t                 aReceiveBufferInUse)
 {
-    OT_UNUSED_VARIABLE(iid);
-    OT_ASSERT(pPacketDetails != NULL);
-
     // Current time > sync-receive timestamp
     // Therefore lower 32 bits of current time should always be greater than lower 32 bits
     // of sync-rx timestamp unless there is a overflow.  In such cases, we do not want to
     // take overflow into consideration for sync-rx timestamp.
-    uint64_t railUsTimeNow    = otPlatTimeGet();
-    uint32_t railUsTimerWraps = railUsTimeNow >> 32;
+    uint64_t    railUsTimeNow    = otPlatTimeGet();
+    uint32_t    railUsTimerWraps = railUsTimeNow >> 32;
+    RAIL_Time_t timestamp        = (pPacketDetails != NULL ? pPacketDetails->timeReceived.packetTime
+                                                           : sReceivePacket[aReceiveBufferInUse].packetInfo.timestamp);
 
     // Address multiple overflows, such as what would happen if the current time overflows
     // from 0x00000001FFFFFFFF to 0x0000000200000000 (leave the higher 32 bits as 0)
-    if ((railUsTimeNow & 0xFFFFFFFF) <= pPacketDetails->timeReceived.packetTime)
+    if ((railUsTimeNow & 0xFFFFFFFF) <= timestamp)
     {
         railUsTimerWraps--;
     }
 
-    if (pPacketDetails->isAck)
+    if (pPacketDetails != NULL && pPacketDetails->isAck)
     {
-        sReceiveAckFrame.mInfo.mRxInfo.mRssi = pPacketDetails->rssi;
-        sReceiveAckFrame.mInfo.mRxInfo.mLqi  = pPacketDetails->lqi;
-        sReceiveAckFrame.mInfo.mRxInfo.mTimestamp =
-            pPacketDetails->timeReceived.packetTime + ((uint64_t)railUsTimerWraps << 32);
-// TODO: Maybe remove #if ?
+        sReceiveAckFrame.mInfo.mRxInfo.mRssi      = pPacketDetails->rssi;
+        sReceiveAckFrame.mInfo.mRxInfo.mLqi       = pPacketDetails->lqi;
+        sReceiveAckFrame.mInfo.mRxInfo.mTimestamp = timestamp + ((uint64_t)railUsTimerWraps << 32);
 #if OPENTHREAD_RADIO && OPENTHREAD_CONFIG_MULTIPAN_RCP_ENABLE == 1
```

Comment:
GSDK version does not have #if wrapper around   `sReceiveAckFrame.mIid = iid; `

Reply:
You can ignore this comment now.  I understand why there is a required difference.